modules/distribution/src/bin/adaptive.sh

```diff
@@ -127,7 +127,11 @@ else
   else
     sever_restart_required=true
     echo "Nashorn library not found. Starting to download....."
-    curl https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar -o $CARBON_HOME/repository/components/lib/nashorn-core-$NASHORN_VERSION.jar
+    res_code=$(curl https://repo1.maven.org/maven2/org/openjdk/nashorn/nashorn-core/$NASHORN_VERSION/nashorn-core-$NASHORN_VERSION.jar -o $CARBON_HOME/repository/components/lib/nashorn-core-$NASHORN_VERSION.jar)
+    if test "$exit_code" != "0"; then
+      echo "Nashorn library download failed with exit code: $exit_code"
+      exit 1
+    fi
```

Comment:
Don't we need to add this changes to the adaptive.bat file?

Reply:
The related issue was reported for only the shell script. Shall we look into bat files as well?